### ISSUE TITLE
Sporadic Build failure due to failed test (testTransaction)

### ISSUE THREAD

**Issue by RaulGracia** [pravega/pravega#2686]
**Problem description**
It has been observed a single test failure during the Jenkins build (984) corresponding to the `rollingTransactions` functionality [merge](https://github.com/pravega/pravega/commit/64600702359cdcdba3c2b5a5d3275c9b9cb4c191). The failed test is:
`io.pravega.segmentstore.server.host.handler.PravegaRequestProcessorTest.testTransaction`.

At first glance, the test fails due to a timeout exception:
```
org.junit.runners.model.TestTimedOutException: test timed out after 20000 milliseconds
    at sun.misc.Unsafe.park(Native Method)
    at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
    at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
```

This may be caused by other exceptions seen in the logs. In particular:

```
2018-06-09 14:04:05,082 WARN  [Time-limited test] handler.PravegaRequestProcessor (PravegaRequestProcessor.java:handleException(430)) - Unsupported Operation 'Create segment'.
java.lang.UnsupportedOperationException: createStreamSegment is unsupported on ReadOnlySegmentContainer
    at io.pravega.segmentstore.server.containers.ReadOnlySegmentContainer.unsupported(ReadOnlySegmentContainer.java:195)
    at io.pravega.segmentstore.server.containers.ReadOnlySegmentContainer.createStreamSegment(ReadOnlySegmentContainer.java:171)
    at io.pravega.segmentstore.server.store.StreamSegmentService.lambda$createStreamSegment$6(StreamSegmentService.java:112)
    at io.pravega.segmentstore.server.store.StreamSegmentService.invoke(StreamSegmentService.java:174)
    at io.pravega.segmentstore.server.store.StreamSegmentService.createStreamSegment(StreamSegmentService.java:110)
```
```
io.pravega.segmentstore.contracts.BadAttributeUpdateException: [Segment 'testSegmentAttribute'] Bad Attribute Update (AttributeId = 67755ffe-6d46-4536-bfda-24faee03b57b, Value = 5, UpdateType = ReplaceIfEquals): Expected existing value to be '-9223372036854775808', actual '1'..
    at io.pravega.segmentstore.server.logs.SegmentMetadataUpdateTransaction.preProcessAttributes(SegmentMetadataUpdateTransaction.java:429)
    at io.pravega.segmentstore.server.logs.SegmentMetadataUpdateTransaction.preProcessOperation(SegmentMetadataUpdateTransaction.java:274)
    at io.pravega.segmentstore.server.logs.ContainerMetadataUpdateTransaction.preProcessOperation(ContainerMetadataUpdateTransaction.java:321
```

**Problem location**
Test: `io.pravega.segmentstore.server.host.handler.PravegaRequestProcessorTest.testTransaction`

**Suggestions for an improvement**
Fix the issue that makes the test to fail.

**Comment by shrids** [comment]
Ran this test(with the changes in PR #3191)  in a loop for the last 12 hours and am not able to simulate this error.

**Comment by andreipaduroiu** [comment]
If this was fixed with #3191, then let's reopen it if it shows up again.
